sdk/core/core-client-rest/src/getClient.ts

```diff
@@ -59,9 +60,13 @@ export function getClient(
   }
 
   const { allowInsecureConnection, httpClient } = clientOptions;
+  const endpointUrl = endpoint ?? clientOptions.endpoint ?? clientOptions.baseUrl;
```

Comment:
now I am doubt if endpoint ever going to be undefined ?

Reply:
You are right, it is never going to be undefined. I think the intent of this originally was to allow overriding the endpoint with the options, but the current code won't do it. It should be like

```ts
let endpointUrl = endpoint;
const endpointOverride = clientOptions.endpoint ?? clientOptions.baseUrl;
if (endpointOverride) {
   endpointUrl = endpointOverride
}
```